pyface/tests/test_widget.py

```diff
@@ -132,6 +158,75 @@ def test_visible(self):
 
         self.assertFalse(self.widget.control.isVisible())
 
+    def test_contents_visible(self):
+        window = MainWindow()
+        window._create()
+
+        try:
+            with self.event_loop():
+                window.open()
+
+            with self.assertTraitDoesNotChange(window.widget, "visible"):
+                with self.event_loop():
+                    window.visible = False
+
+            with self.assertTraitDoesNotChange(window.widget, "visible"):
+                with self.event_loop():
+                    window.visible = True
```

Comment:
This is testing that when you toggle the visibility of the parent, the `visible` trait of the child does not change, which is the desired behaviour.

Reply:
Could you please add this explanation to the test? This being the intent of the test wasn't obvious to me at all when I read the test. If it is not obvious to me, it is likely to be the case for someone else, and it would save them a lot of time puzzling or getting confused if we have just this sentence as a comment. Same for the other tests.